pkgs/development/cuda-modules/cudnn-frontend/default.nix

```diff
@@ -0,0 +1,134 @@
+{
+  autoAddDriverRunpath,
+  catch2_3,
+  cmake,
+  cudaPackages,
+  fetchFromGitHub,
+  gitUpdater,
+  lib,
+  ninja,
+  nlohmann_json,
+  stdenv,
+}:
+let
+  inherit (lib.lists) optionals;
+  inherit (lib.strings)
+    cmakeBool
+    cmakeFeature
+    optionalString
+    ;
+  inherit (cudaPackages)
+    cuda_cccl
+    cuda_cudart
+    cuda_nvcc
+    cuda_nvrtc
+    cudnn
+    libcublas
+    ;
+in
+
+# TODO(@connorbaker): This should be a hybrid C++/Python package.
+stdenv.mkDerivation (finalAttrs: {
+  pname = "cudnn-frontend";
+  version = "1.9.0";
+
+  src = fetchFromGitHub {
+    owner = "NVIDIA";
+    repo = "cudnn-frontend";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-Vc5jqB1XHcJEdKG0nxbWLewW2fDezRVwjUSzPDubSGE=";
+  };
+
+  patches = [
+    # https://github.com/NVIDIA/cudnn-frontend/pull/125
+    ./0001-cmake-float-out-common-python-bindings-option.patch
+    ./0002-cmake-add-config-so-headers-can-be-discovered-when-i.patch
+    ./0003-cmake-install-samples-and-tests-when-built.patch
+    ./0004-samples-fix-instances-of-maybe-uninitialized.patch
+  ];
+
+  # TODO: As a header-only library, we should make sure we have an `include` directory or similar which is not a
+  # superset of the `out` (`bin`) or `dev` outputs (whih is what the multiple-outputs setup hook does by default).
+  outputs =
+    [
+      "out"
+    ]
+    ++ optionals finalAttrs.doCheck [
+      "legacy_samples"
+      "samples"
+      "tests"
+    ];
+
+  nativeBuildInputs = [
+    autoAddDriverRunpath # Needed for samples because it links against CUDA::cuda_driver
+    cmake
+    cuda_cccl
```

Comment:
This only contains headers and cmake files, do we really need this in `native*`?

Reply:
Good catch! Yes, we can put `cuda_cccl` in `buildInputs` -- I just tried building with it that way and it worked.

Reply:
Added a comment, because I'm clueless as well.

Reply:
Moved into buildInputs before I merged it.